modules/objdetect/src/qrcode.cpp

```diff
@@ -2251,6 +2251,91 @@ bool QRDecode::preparingCurvedQRCodes()
     return true;
 }
 
+/**
+ * @param finderPattern 4 points of finder pattern markers, calculated by findPatternsVerticesPoints()
+ * @return true if the pattern has the correct side lengths
+ */
+static inline bool checkFinderPatternByAspect(const vector<Point> &finderPattern) {
+    if (finderPattern.size() != 4ull)
+        return false;
+    vector<float> sidesLen;
+    for (size_t i = 0; i < finderPattern.size(); i++) {
+        sidesLen.push_back(sqrt(normL2Sqr<float>(Point2f(finderPattern[i] - finderPattern[(i+1ull)%finderPattern.size()]))));
+    }
```

Comment:
As soon as you know `sidesLen` size there is no need in `std::vector<>` and `push_back` with possible memory re-allocation. You can use `float sidesLen[4];`

Reply:
fixed